src/plots/mod.rs

```diff
@@ -0,0 +1,26 @@
+pub mod histogram;
+
+use crate::util::Result;
+pub use histogram::Histogram;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+pub trait Plot {
+    type PlotDataMetadataType: Debug + PartialEq + Serialize;
+
+    fn to_vega_embeddable(
+        &self,
+        allow_interactions: bool,
+    ) -> Result<PlotData<Self::PlotDataMetadataType>>;
+
+    fn to_png(&self, width_px: u16, height_px: u16) -> Vec<u8>;
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct PlotData<T>
+where
+    T: Serialize,
```

Comment:
why is it T here and not PlotDataMetadataType?

Reply:
You don't know how the metadata type will look like for a certain plot.
However, you want to have it serializable.

Reply:
yes, but wouldn't it be the same type as in Plot?

Reply:
I matched the requirements in e34c177.
However, PlotDataMetadataType is only an inner type of the trait. Hence, each plot must define it for its needs. It only needs to be serializable.